CMakeLists.txt

```diff
@@ -830,7 +795,7 @@ JOIN("-I${DEPENDENCIES_INCLUDE_DIRS}" " -I" PKGCONFIG_OSRM_INCLUDE_FLAGS)
 # Boost uses imported targets, we need to use a generator expression to extract
 # the link libraries to be written to the pkg-config file.
 foreach(engine_lib ${ENGINE_LIBRARIES})
-  if("${engine_lib}" MATCHES "^Boost.*")
+  if("${engine_lib}" MATCHES "^Boost.*" OR "${engine_lib}" MATCHES "^CONAN_LIB.*")
```

Comment:
What do the`CONAN_LIB.` targets represent?

Reply:
Its CMake targets for dependencies defined by Conan, otherwise we have problems with build of example. It tries to find libraries with name like CONAN_LIB::BZip2 and cannot. I guess we had the same issue with Boost in the past. https://github.com/Project-OSRM/osrm-backend/blob/06b1b980bbdfc838c4f8e8cb370cba6dfb0329b0/example/CMakeLists.txt#L34

I added a comment to make it more clear.